mypyc/codegen/emitfunc.py

```diff
@@ -199,13 +191,50 @@ def visit_load_error_value(self, op: LoadErrorValue) -> None:
             self.emit_line('%s = %s;' % (self.reg(op),
                                          self.c_error_value(op.type)))
 
+    def get_attr_expr(self, obj: str, op: Union[GetAttr, SetAttr], decl_cl: ClassIR) -> str:
+        """Generate attribute accessor for normal (non-property) access.
+
+        This either has a form like obj->attr_name for attributes defined in non-trait
+        classes, and *(obj + attr_offset) for attributes defined by traits. We also
+        insert all necessary C casts here.
+        """
+        cast = '({} *)'.format(op.class_type.struct_name(self.emitter.names))
+        if decl_cl.is_trait and op.class_type.class_ir.is_trait:
+            # For pure trait access find the offset first, offsets
+            # are ordered by attribute position in the cl.attributes dict.
+            trait_attr_index = list(decl_cl.attributes).index(op.attr)
```

Comment:
Put a `TODO` here or something so we can find this easy if we ever want to improve the perf